hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs

```diff
@@ -245,17 +245,17 @@ setCurrentCradle cradle = do
     let ps = mapMaybe (GHC.ml_hs_file . GHC.ms_location) (mgModSummaries mg)
     debugm $ "Modules in the cradle: " ++ show ps
     ps' <- liftIO $ mapM canonicalizePath ps
-    modifyCache (\s -> s { currentCradle = Just (ps', cradle) })
+    modifyCache (\s -> s { currentCradle = Just (ps', cradle, Nothing) })
```

Comment:
This should definitely have an option to cache the `ComponentOptions` here. You can obtain the ComponentOptions by splitting the call to `BIOS.initializeFlagsWithCradleWithMessage` into its implementation https://github.com/mpickering/hie-bios/blob/master/src/HIE/Bios/Ghc/Api.hs#L79. Ideally, we would expose that somehow in the API, maybe make an issue upstream?

Reply:
That does seem like better approach